lein.bat plugin development issue below windows, refer Issue #718

Hi,
I found there still have problem with lein.bat with .lein-classpath under windows.
OS: Windows Vista
Editor: Light Table

.lein-class
..\myplugin\src\

Error:
Exception in thread "main" java.lang.NoClassDefFoundError: ;C:\Users\LiuHao\Programs\leiningen\self-installs\leiningen-2/2/0-standalone/
Caused by: java.lang.ClassNotFoundException: ;C:\Users\LiuHao\Programs\leiningen\self-installs\leiningen-2.2.0-standalone.jar
        at java.net.URLClassLoader$1.run(Unknown Source)
        at java.security.AccessController.doPrivileged(Native Method)
        at java.net.URLClassLoader.findClass(Unknown Source)
        at java.lang.ClassLoader.loadClass(Unknown Source)
        at sun.misc.Launcher$AppClassLoader.loadClass(Unknown Source)
        at java.lang.ClassLoader.loadClass(Unknown Source)
Could not find the main class: ;C:\Users\LiuHao\Programs\leiningen\self-installs\leiningen-2.2.0-standalone.jar.  Program will exit.

The error is caused by trailing space from classpath, but I am not sure where it come from.

Suggest change, just trim the string:
from 
for /f %%i in (".lein-classpath") do set CONTEXT_CP=%%i 
to
for /f "tokens=* delims= " %%i in (.lein-classpath) do set CONTEXT_CP=%%i

--- a/bin/lein.bat
+++ b/bin/lein.bat
@@ -101,7 +101,9 @@ rem the paths inside the bootstrap file do not already contain double quotes but
 
     :: Apply context specific CLASSPATH entries
     if exist "%~dp0..\.lein-classpath" (
-        for /f %%i in ("%~dp0...lein-classpath") do set CONTEXT_CP=%%i
+        for /f "tokens=* delims= " %%i in (%~dp0..\.lein-classpath) do (
+            set CONTEXT_CP=%%i
+        )
 
         if NOT "x!CONTEXT_CP!"=="x" (
             set CLASSPATH=!CONTEXT_CP!;!CLASSPATH!
@@ -114,7 +116,9 @@ rem the paths inside the bootstrap file do not already contain double quotes but
     set CLASSPATH=%LEIN_JAR%
   
     if exist ".lein-classpath" (
-        for /f %%i in (.lein-classpath) do set CONTEXT_CP=%%i 
+        for /f "tokens=* delims= " %%i in (.lein-classpath) do (
+            set CONTEXT_CP=%%i
+        )
 
         if NOT "x!CONTEXT_CP!"=="x" (
             set CLASSPATH=!CONTEXT_CP!;!CLASSPATH!

